bittide-instances/src/Bittide/Instances/Hitl/Post/TcpServer.hs

```diff
@@ -0,0 +1,52 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+{-# LANGUAGE OverloadedStrings #-}
+
+module Bittide.Instances.Hitl.Post.TcpServer where
+
+import Data.Time.Clock.POSIX
+import System.IO
+import Prelude
+
+import qualified Network.Simple.TCP as NS
+import qualified Streaming.ByteString as SBS
+
+startServer :: IO (NS.Socket, NS.SockAddr)
+startServer = do
+  (serverSock, serverAddr) <- NS.bindSock (NS.Host "0.0.0.0") "1234"
+  let host = (serverSock, serverAddr)
+  putStrLn $ "Listening for connections on " <> show serverAddr
+  NS.listenSock serverSock 2048
+  pure host
+
+waitForClients :: Int -> NS.Socket -> IO [(NS.Socket, NS.SockAddr)]
+waitForClients numberOfClients serverSock = do
+  mapM
+    ( \i ->
+        NS.accept
+          serverSock
+          ( \(clientSock, clientAddr) -> do
+              putStrLn $ show i <> " | Connection established from: " ++ show clientAddr
+              pure (clientSock, clientAddr)
+          )
+    )
+    [1 .. numberOfClients]
+
+clientDataToFile :: Handle -> Int -> NS.Socket -> IO ()
+clientDataToFile handle bufferSize sock = do
+  start <- getPOSIXTime
+  SBS.toHandle handle $ SBS.reread (const $ NS.recv sock bufferSize) ()
+  stop <- getPOSIXTime
+  size <- hFileSize handle
+  let speed = fromIntegral size / (stop - start)
```

Comment:
You need some extra type conversion here.
Because now `speed :: NominalDiffTime`, and it will be shown with a "s" suffix, because it is supposed to represent seconds.

Reply:
If you delete `clientDataToFile` then you can ignore this ^